enterprise/frontend/src/embedding-sdk/hooks/private/use-load-question.ts

```diff
@@ -79,23 +82,39 @@ export function useLoadQuestion({
   // Avoid re-running the query if the parameters haven't changed.
   const sqlParameterKey = getParameterDependencyKey(initialSqlParameters);
 
-  const [loadQuestionState, loadQuestion] = useAsyncFn(async () => {
-    const state = await dispatch(
-      runQuestionOnLoadSdk({
+  const shouldLoadQuestion = cardId != null;
+  const [isQuestionLoading, setIsQuestionLoading] =
+    useState(shouldLoadQuestion);
+
+  const [, loadAndQueryQuestion] = useAsyncFn(async () => {
+    if (shouldLoadQuestion) {
+      setIsQuestionLoading(true);
+    }
+    const questionState = await dispatch(
+      loadQuestionSdk({
         options,
         deserializedCard,
         cardId,
-        cancelDeferred: deferred(),
         initialSqlParameters,
       }),
-    );
+    ).finally(() => {
+      setIsQuestionLoading(false);
```

Comment:
Set the loading to false whether this promise is resolved or not. It will be rejected if the the provided ID couldn't be loaded.

This should fix #51969